aws/resource_aws_launch_template_test.go

```diff
@@ -0,0 +1,228 @@
+					resource.TestCheckResourceAttrSet(resName, "key_name"),
+					resource.TestCheckResourceAttrSet(resName, "monitoring"),
+					resource.TestCheckResourceAttr(resName, "network_interfaces.#", "1"),
+					resource.TestCheckResourceAttr(resName, "placement.#", "1"),
+					resource.TestCheckResourceAttrSet(resName, "ram_disk_id"),
+					resource.TestCheckResourceAttr(resName, "vpc_security_group_ids.#", "1"),
+					resource.TestCheckResourceAttr(resName, "tag_specifications.#", "1"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSLaunchTemplate_update(t *testing.T) {
+	var template ec2.LaunchTemplate
+	resName := "aws_launch_template.foo"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSLaunchTemplateDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSLaunchTemplateConfig_basic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSLaunchTemplateExists(resName, &template),
+					resource.TestCheckResourceAttr(resName, "default_version", "1"),
+					resource.TestCheckResourceAttr(resName, "latest_version", "1"),
+				),
+			},
+			{
+				Config: testAccAWSLaunchTemplateConfig_data,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSLaunchTemplateExists(resName, &template),
+					resource.TestCheckResourceAttr(resName, "default_version", "1"),
+					resource.TestCheckResourceAttr(resName, "latest_version", "2"),
+					resource.TestCheckResourceAttrSet(resName, "image_id"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSLaunchTemplateExists(n string, t *ec2.LaunchTemplate) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No Launch Template ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).ec2conn
+
+		resp, err := conn.DescribeLaunchTemplates(&ec2.DescribeLaunchTemplatesInput{
+			LaunchTemplateIds: []*string{aws.String(rs.Primary.ID)},
+		})
+		if err != nil {
+			return err
+		}
+
+		if len(resp.LaunchTemplates) != 1 || *resp.LaunchTemplates[0].LaunchTemplateId != rs.Primary.ID {
+			return fmt.Errorf("Launch Template not found")
+		}
+
+		*t = *resp.LaunchTemplates[0]
+
+		return nil
+	}
+}
+
+func testAccCheckAWSLaunchTemplateDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).ec2conn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_launch_template" {
+			continue
+		}
+
+		resp, err := conn.DescribeLaunchTemplates(&ec2.DescribeLaunchTemplatesInput{
+			LaunchTemplateIds: []*string{aws.String(rs.Primary.ID)},
+		})
+
+		if err == nil {
+			if len(resp.LaunchTemplates) != 0 && *resp.LaunchTemplates[0].LaunchTemplateId == rs.Primary.ID {
+				return fmt.Errorf("Launch Template still exists")
+			}
+		}
+
+		ae, ok := err.(awserr.Error)
```

Comment:
This can be simplified with `if isAWSErr(err, "InvalidLaunchTemplateId.NotFound", "") {` 👍  (replacing the middle with an SDK constant if available)